src/System.Memory/src/System/Buffers/ReadOnlySequence.cs

```diff
@@ -385,15 +385,23 @@ private ReadOnlySequence<T> SliceImpl(in SequencePosition begin, in SequencePosi
             );
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private SequenceType GetSequenceType()
+        {
+            // We take high order bits of two indexes and move them
+            // to a first and second position to convert to SequenceType
+            return (SequenceType)(-(2 * (_sequenceStart.GetInteger() >> 31) + (_sequenceEnd.GetInteger() >> 31)));
```

Comment:
would it make sense to cast to uint and then shift so that negating is not needed?

Reply:
nit: It might make sense to split this to two lines or create an inline function here - quite a bit of brackets here

Reply:
Early I made so, but @ahsonkhan show that compiler generates strange IL code for `uint`. It is @ahsonkhan  hack.

Reply:
Aside from potentially formatting it more cleanly, I am fine with this change. It is required for performance.

Reply:
We have just checked with @ahsonkhan and it seems the uint version is negligbly shorter and slightly cleaner. He will consider sending PR with that change separately